Description: increases rsync batching size, leaves off troublesome -I rsync
 flag, auto-flushes, and disables motd display during rsync.
Author: Kees Cook <kees@ubuntu.com>
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=455082

Index: debmirror-2.4.4/debmirror
===================================================================
--- debmirror-2.4.4.orig/debmirror	2010-03-25 11:13:29.000000000 -0700
+++ debmirror-2.4.4/debmirror	2010-04-05 11:36:47.764802267 -0700
@@ -311,7 +311,7 @@
 =item --rsync-options=options
 
 Specify alternative rsync options to be used. Default options are
-"-aIL --partial". Care must be taken when specifying alternative
+"-aL --partial". Care must be taken when specifying alternative
 options not to disrupt operations, it's best to only add to those
 options.
 
@@ -542,7 +542,7 @@
 our $download_method="ftp";
 our $timeout=300;
 our $max_batch=0;
-our $rsync_batch=200;
+our $rsync_batch=300;
 our $num_errors=0;
 our $bytes_to_get=0;
 our $bytes_gotten=0;
@@ -553,7 +553,7 @@
 our $start_time = time;
 our $dry_run=0;
 our $do_dry_run=0;
-our $rsync_options="-aIL --partial";
+our $rsync_options="-aL --partial";
 our $ignore_small_errors=0;
 our $diff_mode="use";
 our $gzip_options="-9 -n --rsyncable";
@@ -563,6 +563,9 @@
 my $HOME;
 ($HOME = $ENV{'HOME'}) or die "HOME not defined in environment!\n";
 
+# Switch to auto-flushing mode for stdout.
+select STDOUT; $|=1;
+
 # Load in config files
 require "/etc/debmirror.conf" if -r "/etc/debmirror.conf";
 require "$HOME/.debmirror.conf" if -r "$HOME/.debmirror.conf";
@@ -1267,7 +1270,7 @@
 	      print $fh "$_\n";
 	    }
 	  }
-	  system ("rsync --timeout=$timeout $opt $rsyncremote --include-from=$rsynctempfile --exclude='*' $mirrordir");
+	  system ("rsync --no-motd --timeout=$timeout $opt $rsyncremote --include-from=$rsynctempfile --exclude='*' $mirrordir");
 	  close $fh;
 	  unlink $rsynctempfile;
 	  foreach my $dest (@result) {
@@ -1691,7 +1694,7 @@
   }
   $opt = "$opt --progress" if $progress;
   $opt = "$opt -v" if $debug;
-  system ("rsync --timeout=$timeout $opt $rsyncremote --include=$file --exclude='*' .");
+  system ("rsync --no-motd --timeout=$timeout $opt $rsyncremote --include=$file --exclude='*' .");
   if ($? == 0 && -f $file) {
     return 1;
   } else {
@@ -1769,7 +1772,7 @@
   for (@result) {
     print $fh "$_\n";
   }
-  system ("rsync --timeout=$timeout $opt $rsyncremote --delete --include-from=$rsynctempfile --exclude='*' $mirrordir");
+  system ("rsync --no-motd --timeout=$timeout $opt $rsyncremote --delete --include-from=$rsynctempfile --exclude='*' $mirrordir");
   close $fh;
   unlink $rsynctempfile;
 }
